src/manage_nsfs/nc_lifecycle.js

```diff
@@ -170,16 +181,33 @@ async function process_bucket(config_fs, bucket_name, system_json) {
  */
 async function process_rules(config_fs, bucket_json, object_sdk, should_notify) {
     try {
-        await P.all(_.map(bucket_json.lifecycle_configuration_rules,
-            async (lifecycle_rule, index) =>
-                await _call_op_and_update_status({
-                    bucket_name: bucket_json.name,
-                    rule_id: lifecycle_rule.id,
-                    op_name: TIMED_OPS.PROCESS_RULE,
-                    op_func: async () => process_rule(config_fs, lifecycle_rule, index, bucket_json, object_sdk, should_notify)
-                })
-            )
-        );
+        const bucket_state = lifecycle_run_status.buckets_statuses[bucket_json.name].state || {};
+        let num_objects_processed = 0;
+        while (!bucket_state.is_finished && num_objects_processed < config.NC_LIFECYCLE_BUCKET_BATCH_SIZE) {
+            dbg.log0("process_rules");
```

Comment:
remove log or change it to log2 + something more informative

Reply:
I removed it entirely